node/forest_libp2p/src/service.rs

```diff
@@ -321,12 +326,44 @@ where
                         NetworkMessage::JSONRPCRequest { method } => {
                             match method {
                                 NetRPCMethods::NetAddrsListen(response_channel) => {
-                                let listeners: Vec<_> = Swarm::listeners( swarm_stream.get_mut()).cloned().collect();
-                                let peer_id = Swarm::local_peer_id(swarm_stream.get_mut());
+                                    let listeners: Vec<_> = Swarm::listeners( swarm_stream.get_mut()).cloned().collect();
+                                    let peer_id = Swarm::local_peer_id(swarm_stream.get_mut());
+
                                     if response_channel.send((*peer_id, listeners)).is_err() {
                                         warn!("Failed to get Libp2p listeners");
                                     }
                                 }
+                                NetRPCMethods::NetPeers(response_channel) => {
+                                    let peer_addresses: &HashMap<PeerId, Vec<Multiaddr>> = swarm_stream.get_mut().peer_addresses();
+
+                                    if response_channel.send(peer_addresses.to_owned()).is_err() {
+                                        warn!("Failed to get Libp2p peers");
+                                    }
+                                }
+                                NetRPCMethods::NetConnect(response_channel, peer_id, addresses) => {
+                                    let mut addresses = addresses.clone();
+                                    let mut success = false;
+
+                                    for multiaddr in addresses.iter_mut() {
+                                        multiaddr.push(Protocol::P2p(Multihash::from_bytes(&peer_id.to_bytes()).unwrap()));
```

Comment:
Does this need a `cargo fmt`? Hopefully the CI will say if so

Reply:
No, `cargo fmt` suggests no changes.

Reply:
Yeah, seems like it was a Github rendering issue, not an issue with the actual file

Reply:
Gotcha